android/src/main/java/com/twilio/twilio_voice/IncomingCallNotificationService.java

```diff
@@ -140,14 +143,28 @@ private Notification buildNotification(String title, String text, PendingIntent
         rejectIntent.setAction(Constants.ACTION_REJECT);
         rejectIntent.putExtra(Constants.INCOMING_CALL_INVITE, callInvite);
         rejectIntent.putExtra(Constants.INCOMING_CALL_NOTIFICATION_ID, notificationId);
-        PendingIntent piRejectIntent = PendingIntent.getService(getApplicationContext(), 0, rejectIntent, PendingIntent.FLAG_UPDATE_CURRENT);
-
-        Intent acceptIntent = new Intent(getApplicationContext(), IncomingCallNotificationService.class);
-        acceptIntent.setAction(Constants.ACTION_ACCEPT);
-        acceptIntent.putExtra(Constants.ACCEPT_CALL_ORIGIN, 0);
-        acceptIntent.putExtra(Constants.INCOMING_CALL_INVITE, callInvite);
-        acceptIntent.putExtra(Constants.INCOMING_CALL_NOTIFICATION_ID, notificationId);
-        PendingIntent piAcceptIntent = PendingIntent.getService(getApplicationContext(), 0, acceptIntent, PendingIntent.FLAG_UPDATE_CURRENT);
+        PendingIntent piRejectIntent = PendingIntent.getService(getApplicationContext(), 0, rejectIntent, PendingIntent.FLAG_UPDATE_CURRENT |  PendingIntent.FLAG_IMMUTABLE);
+
+        Intent acceptIntent;
+        PendingIntent piAcceptIntent;
+        // VERSION S = Android 12
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.S) {
```

Comment:
Nice so Version S+ we launch an activity which will launch this service again

Any version lower than that will perform a 'notification trampoline' and will launch the service (itself) directly from here